drivers/adc/ad9680/ad9680.c

```diff
@@ -109,6 +115,114 @@ int32_t ad9680_test(struct ad9680_dev *dev,
 	return(0);
 }
 
+static int ad9680_jesd204_link_init(struct jesd204_dev *jdev,
+				    enum jesd204_state_op_reason reason,
+				    struct jesd204_link *lnk)
+{
+	struct ad9680_jesd204_priv *priv = jesd204_dev_priv(jdev);
+	struct ad9680_dev *dev = priv->dev;
+	struct jesd204_link *link;
+
+	switch (reason) {
+	case JESD204_STATE_OP_REASON_INIT:
+		break;
+	default:
+		return JESD204_STATE_CHANGE_DONE;
+	}
+
+	pr_debug("%s:%d link_num %u reason %s\n", __func__,
+		 __LINE__, lnk->link_id, jesd204_state_op_reason_str(reason));
+
+	link = &dev->jesd204_link;
+
+	jesd204_copy_link_params(lnk, link);
+
+	lnk->sample_rate = dev->sampling_frequency_hz;
+	lnk->sample_rate_div = dev->dcm;
+	lnk->jesd_encoder = JESD204_ENCODER_8B10B;
+
+	if (dev->sysref_mode == AD9680_SYSREF_CONT)
+		lnk->sysref.mode = JESD204_SYSREF_CONTINUOUS;
+	else if (dev->sysref_mode == AD9680_SYSREF_ONESHOT)
+		lnk->sysref.mode = JESD204_SYSREF_ONESHOT;
+
+	return JESD204_STATE_CHANGE_DONE;
+}
+
+int ad9680_jesd_enable_link(struct ad9680_dev *dev, uint8_t en)
+{
+	int err;
+	uint8_t tmp_reg;
+
+	if (dev == NULL)
+		return -11;
```

Comment:
Why are we returning the explicit error code instead of a macro?

Reply:
This has been preserved for sync with Linux variant. Replaced by ENOMEM and EINVAL.